specs/executionAPISpec.md

```diff
@@ -0,0 +1,530 @@
+* [EC-19] eth_call **MUST** return 0x0 when using block tags for `defaultBlockParameter` while syncing to the network.
+## <p id="fillTransaction">eth_fillTransaction</p>
+* [EFT-1] eth_fillTransaction **MUST** fill in the missing transaction parameters of the given transaction.
+* [EFT-2] eth_fillTransaction **MUST** return the return the raw transaction and JSON transaction object of the filled transaction.
+* [EFT-3] eth_filterTransaction **MUST NOT** sign the transaction.
+* [EFT-4] eth_filterTransaction **MUST** error with code -32000 when the `from` address does not have enough Ether to pay for the transaction.
+* [EFT-5] eth_filterTransaction **MUST** allow both `data` and `input` to be used for contract creation,
+* [EFT-6] eth_filterTransaction **MUST** error with code -32000 when the `data` and `input` both specified and not equal
+* [EFT-7] eth_filterTransaction **MUST** error with code -32000 when `data` or `input` or `to` is not specified.
+* [EFT-8] eth_filterTransaction **MUST** error with code -3200 when `data` or `input` provided caused an EVM error when the `gas` is not specified, otherwise it is ignored.
+## <p id="createAccessList">eth_createAccessList</p>
+* [ECAL-1] eth_createAccessList **MUST** return an [EIP-2930](https://eips.ethereum.org/EIPS/eip-2930) access list biased off the given `transaction` and estimated gas cost when using the access list in the `transaction`.
+* [ECAL-2] eth_createAccessList **MUST** use "latest" when the `defaultBlockParameter` is not specified.
+* [ECAL-3] eth_createAccessList **MUST** use the estimated gas when the `gas` is not specified.
+* [ECAL-4] eth_createAccessList **MUST** error with code -32000 when `gas` is too low.
+* [ECAL-5] eth_createAccessList **MUST** error with code -32000 when the `from` address does not have enough Ether to pay for the transaction.
+## <p id="estimateGas">eth_estimateGas</p>
+* [EEG-1] eth_estimateGas **MUST** return the estimated amount of gas the given `transaction` will take to execute.
+* [EEG-2] eth_estimateGas **MUST** error with code -32000 when the `from` address has insufficient Ether to execute the given `transaction`.
+* [EEG-3] eth_estimateGas **MUST NOT** check if the `from` account has sufficient funds when estimating contract deployment.
+* [EEG-4] eth_estimateGas **MUST** use 0x0000000000000000000000000000000000000000 for `from` when it is not specified.
+* [EEG-5] eth_estimateGas **MUST** error with code -32000 when estimating a contract creation that causes an error within the EVM.
+## <p id="signTransaction">eth_signTransaction</p>
+* [ESNT-1] eth_signTransaction **MUST** return both the signed encoded and signed unencoded format of the `transaction` given.
+* [ESNT-2] eth_signTransaction **MUST NOT** use given transaction `type` for transaction type, **MUST** determine transaction type from given transaction parameters.
```

Comment:
This seems very wrong.  If the user specifies `type: 2` in their transaction, a type 2 transaction **MUST** be signed IMO.

If this isn't happening in current clients, we should file a bug with those clients.  It may be valuable to indicate in this specification that if the parameters provided don't align with the transaction type provided by the caller then an error should be returned (with details about that error).

Reply:
Right now it determines the transaction type by how you pay for gas. when `gasPrice` is used it defaults to Legacy, and when `maxFeePerGas` and `maxPriorityFeePerGas` are used it defaults to type 2 transaction. I will open up a geth issue.

Reply:
In Geth, if you supply a `type` parameter does it just ignore it completely?

Reply:
Here is the Geth [issue](https://github.com/ethereum/go-ethereum/issues/24179). Apparently Geth doesn't have a `type` field at all.